packages/flutter/lib/src/painting/edge_insets.dart

```diff
@@ -191,52 +364,36 @@ class EdgeInsets {
     return new Rect.fromLTRB(rect.left + left, rect.top + top, rect.right - right, rect.bottom - bottom);
   }
 
-  /// Returns a new size that is bigger than the given size by the amount of
-  /// inset in the horizontal and vertical directions.
-  ///
-  /// See also:
-  ///
-  ///  * [inflateRect], to inflate a [Rect] rather than a [Size].
-  ///  * [deflateSize], to deflate a [Size] rather than inflating it.
-  Size inflateSize(Size size) {
-    return new Size(size.width + horizontal, size.height + vertical);
-  }
-
-  /// Returns a new size that is smaller than the given size by the amount of
-  /// inset in the horizontal and vertical directions.
-  ///
-  /// If the argument is smaller than [collapsedSize], then the resulting size
-  /// will have negative dimensions.
-  ///
-  /// See also:
-  ///
-  ///  * [deflateRect], to deflate a [Rect] rather than a [Size].
-  ///  * [inflateSize], to inflate a [Size] rather than deflating it.
-  Size deflateSize(Size size) {
-    return new Size(size.width - horizontal, size.height - vertical);
-  }
-
-  /// Returns the difference between two EdgeInsets.
-  EdgeInsets operator -(EdgeInsets other) {
-    return new EdgeInsets.fromLTRB(
-      left - other.left,
-      top - other.top,
-      right - other.right,
-      bottom - other.bottom
-    );
+  /// Returns the difference between two [EdgeInsets].
+  @override
+  EdgeInsetsGeometry operator -(EdgeInsetsGeometry other) {
+    if (other is EdgeInsets) {
+      return new EdgeInsets.fromLTRB(
+        left - other.left,
+        top - other.top,
+        right - other.right,
+        bottom - other.bottom
+      );
+    }
+    return super - other;
   }
 
-  /// Returns the sum of two EdgeInsets.
-  EdgeInsets operator +(EdgeInsets other) {
-    return new EdgeInsets.fromLTRB(
-      left + other.left,
-      top + other.top,
-      right + other.right,
-      bottom + other.bottom
-    );
+  /// Returns the sum of two [EdgeInsets].
+  @override
+  EdgeInsetsGeometry operator +(EdgeInsetsGeometry other) {
```

Comment:
The problem is that this won't work once we turn off implicit downcasts. It's not sound.

(The patch here is a superset of the structure you're describing. The FooDirectional and _MixedFoo classes are merely a detail to make it so that "start" and "left" never appear together on a class you can construct.)